src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php

```diff
@@ -238,9 +238,7 @@ protected function render($view, array $parameters = array(), Response $response
             throw new \LogicException('You can not use the "render" method if the Templating Component or the Twig Bundle are not available.');
         }
 
-        if (null === $response) {
-            $response = new Response();
-        }
+        $response = new Response();
```

Comment:
That's wrong. You can pass a non-null Response as a third parameter in which case your code would override it.